rails_event_store_active_record/lib/rails_event_store_active_record/event_repository.rb

```diff
@@ -146,10 +117,61 @@ def get_all_streams
 
     attr_reader :mapper
 
+    def add_to_stream(collection, stream_name, expected_version, include_global, &to_event_id)
+      raise RubyEventStore::InvalidExpectedVersion if stream_name.eql?(RubyEventStore::GLOBAL_STREAM) && !expected_version.equal?(:any)
+
+      collection = normalize_to_array(collection)
+      expected_version =
+        case expected_version
+        when Integer, :any
+          expected_version
+        when :none
+          -1
+        when :auto
+          eis = EventInStream.where(stream: stream_name).order("position DESC").first
+          (eis && eis.position) || -1
+        else
+          raise RubyEventStore::InvalidExpectedVersion
+        end
+
+      ActiveRecord::Base.transaction(requires_new: true) do
+        in_stream = collection.flat_map.with_index do |element, index|
+          position = unless expected_version.equal?(:any)
+            expected_version + index + POSITION_SHIFT
+          end
+          event_id = to_event_id.call(element)
+          collection = []
+          collection.unshift({
+            stream: RubyEventStore::GLOBAL_STREAM,
+            position: nil,
+            event_id: event_id
+          }) if include_global
+          collection.unshift({
+            stream:   stream_name,
+            position: position,
+            event_id: event_id
+          }) unless stream_name.eql?(RubyEventStore::GLOBAL_STREAM)
+          collection
+        end
+        EventInStream.import(in_stream)
+      end
+      self
+    rescue ActiveRecord::RecordNotUnique => e
+      if detect_pkey_index_violated(e)
+        raise RubyEventStore::EventDuplicatedInStream
+      end
+      raise RubyEventStore::WrongExpectedEventVersion
+    end
+
     def detect_pkey_index_violated(e)
-      e.message.include?("for key 'PRIMARY'")       ||  # MySQL
-      e.message.include?("event_store_events_pkey") ||  # Postgresql
-      e.message.include?("event_store_events.id")       # Sqlite3
+      m = e.message
+      m.include?("for key 'PRIMARY'")       ||  # MySQL
```

Comment:
please consider here naming matching strings as:
```ruby
MYSQL_PRIMARY_KEY_MESSAGE = "for key 'PRIMARY'"
POSTGRES_PRIMARY_KEY_MESSAGE  = "event_store_events_pkey"
...
m.include?(MYSQL_PRIMARY_KEY) ||
m.include?(POSTGRES_PRIMARY_KEY_MESSAGE) 
```

comments will be not needed if constant name will show meaning and intent.

Reply:
Makes sense